examples/deadline/EC2-Image-Builder/python/package/lib/deadline_image.py

```diff
@@ -0,0 +1,243 @@
+            components=component_arn_list,
+            name=f"DeadlineInstallationRecipe{construct_id}",
+            parent_image=props.parent_ami,
+            version=props.image_version,
+        )
+        image_recipe.add_depends_on(deadline_component)
+        for component in props.components:
+            image_recipe.add_depends_on(component)
+
+        infrastructure_configuration = self.create_default_infrastructure_config(scope, construct_id)
+
+        # Create an image using the recipe
+        deadline_image = CfnImage(
+            scope,
+            f"DeadlineImage{construct_id}",
+            image_recipe_arn=image_recipe.attr_arn,
+            infrastructure_configuration_arn=infrastructure_configuration.attr_arn,
+            tags={
+                "DeadlineVersion": props.deadline_version,
+                "ParentAmi": props.parent_ami,
+            },
+        )
+
+        self.ami_id = Token.as_string(deadline_image.get_att("ImageId"))
+        self.node.default_child = deadline_image
+
+    def get_deadline_component_doc(
+        self,
+        deadline_version: str,
+        construct_id: str,
+        os_type: OSType,
+        scope: Construct,
+    ) -> Asset:
+        """
+        Create the YAML document that has the instructions to install Deadline.
+        """
+        if os_type is OSType.LINUX:
+            s3_uri = f"s3://thinkbox-installers/Deadline/{deadline_version}/Linux/DeadlineClient-{deadline_version}-linux-x64-installer.run"
+        else:
+            s3_uri = f"s3://thinkbox-installers/Deadline/{deadline_version}/Windows/DeadlineClient-{deadline_version}-windows-installer.exe"
+
+        return Asset(
+            scope,
+            f"DeadlineComponentDoc{construct_id}",
+            path= template.template_component(
+                props=template.TemplateProps(
+                    template_path=os.path.join(
+                        os.getcwd(),
+                        "..",
+                        "components",
+                        f"deadline-{os_type.value.lower()}.component.template",
+                    ),
+                    tokens={
+                        "s3uri": s3_uri,
+                        "version": deadline_version,
+                    }
+                )
+            )
+        )
+
+    def create_default_infrastructure_config(self, scope: Construct, construct_id: str) -> CfnInfrastructureConfiguration:
+        """
+        Create the default infrastructure config, which defines the permissions needed by Image Builder during
+        image creation.
+        """
+        image_builder_role_name = f"DeadlineImageBuilderRole{construct_id}"
+        image_builder_role = Role(
+            scope,
+            f"DeadlineImageBuilderRole{construct_id}",
+            assumed_by=ServicePrincipal("ec2.amazonaws.com"),
+            role_name= image_builder_role_name,
+        )
+        image_builder_role.add_to_policy(PolicyStatement(
+            actions=[
```

Comment:
Are these actions documented somewhere as required permissions for image builder role?

Reply:
Switched to using Amazon managed policies.